src/index.tsx

```diff
@@ -5,6 +5,49 @@ import {Props, State, Params, Query, Size} from './interfaces';
 import ContainerQueryCore from './ContainerQueryCore';
 import isShallowEqual from './isShallowEqual';
 
+/*
+* const MyComponent = () => {
+*   const [params, containerRef] = useContainerQuery(query);
+*   return <div ref={containerRef} className={classnames(params)}>the box</div>;
+* };
+* SEE: https://github.com/react-container-query/react-container-query/issues/92 for full code explaination
+*/
+
+export const useContainerQuery = (query: Query, initialSize: Size) => {
+  // setup a ref callback
+  // (end user) attaches that ref callback to a container element
+  // @ts-ignore
+  const [params, setParams] = React.useState(() => {
+    if (!initialSize) {
+      return {};
+    }
+    return matchQueries(query)(initialSize);
+  });
+  // @ts-ignore
```

Comment:
The ts-ignore is ugly, but I think we'll have to update typescript and/or types/react, types/react-dom to get it to work.

That should probably be outside the scope of this change, so I'll try to have a look at it later.